### DISCUSSION TITLE
Tuya TS0004

### DISCUSSION THREAD

**Discussion by evangelo18** [Koenkk/zigbee2mqtt#13789] [Devices]
I have a TS0004 4 button Tuya switch. When changing the on/off state to one of the switches from Home Assistant it activates ALL 4 switches instead of just the one.  Anyone else having this issue?
database entry is 
{"id":5,"type":"Router","ieeeAddr":"0xa4c138e161a0a1ae","nwkAddr":61923,"manufId":4417,"manufName":"_TZ3000_ay8hciaf","powerSource":"Mains (single phase)","modelId":"TS0004","epList":[1,2,3,4,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":256,"inClusterList":[3,4,5,6,57344,57345,0],"outClusterList":[25,10],"clusters":{"57344":{"attributes":{"53251":"AAAAAgAABAAABgAA"}},"genBasic":{"attributes":{"65503":"%\u0000\u0000\u0000\u00120\u0000\u0000\u0000\u0016","stackVersion":0,"dateCode":""}},"genOnOff":{"attributes":{"32768":0,"onOff":0,"onTime":0,"offWaitTime":0,"tuyaBacklightMode":1,"moesStartUpOnOff":2,"tuyaBacklightSwitch":1}},"manuSpecificTuya_3":{"attributes":{"powerOnBehavior":2,"switchType":0}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b0024c93867","endpointID":1}],"configuredReportings":[],"meta":{}},"2":{"profId":260,"epId":2,"devId":256,"inClusterList":[4,5,6,57345],"outClusterList":[],"clusters":{"genOnOff":{"attributes":{"onOff":0,"onTime":0,"offWaitTime":0}},"manuSpecificTuya_3":{"attributes":{"powerOnBehavior":2}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b0024c93867","endpointID":1}],"configuredReportings":[],"meta":{}},"3":{"profId":260,"epId":3,"devId":256,"inClusterList":[4,5,6,57345],"outClusterList":[],"clusters":{"genOnOff":{"attributes":{"onOff":0,"onTime":0,"offWaitTime":0}},"manuSpecificTuya_3":{"attributes":{"powerOnBehavior":2}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b0024c93867","endpointID":1}],"configuredReportings":[],"meta":{}},"4":{"profId":260,"epId":4,"devId":256,"inClusterList":[4,5,6,57345],"outClusterList":[],"clusters":{"genOnOff":{"attributes":{"onOff":0,"onTime":0,"offWaitTime":0}},"manuSpecificTuya_3":{"attributes":{"powerOnBehavior":2}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b0024c93867","endpointID":1}],"configuredReportings":[],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":82,"stackVersion":0,"hwVersion":1,"dateCode":"","zclVersion":3,"interviewCompleted":true,"meta":{"configured":1},"lastSeen":1662021873788,"defaultSendRequestWhen":"immediate"}

**Comment by evangelo18** [comment]
Hi anyone having this issue?

**Reply by Koenkk** [reply]
Can you check if this issue has already been fixed in the dev branch? (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

If not; what model shows up in the z2m frontend -> device page?

**Comment by evangelo18** [comment]
The model number that shows up is TS0004.
Manufacturer is  _TZ3000_ay8hciaf
